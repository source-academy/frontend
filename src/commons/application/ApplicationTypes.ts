import { Chapter, Language, SourceError, Variant } from 'js-slang/dist/types';

import { AcademyState } from '../../features/academy/AcademyTypes';
import { AchievementState } from '../../features/achievement/AchievementTypes';
import { DashboardState } from '../../features/dashboard/DashboardTypes';
import { Grading } from '../../features/grading/GradingTypes';
import { PlaygroundState } from '../../features/playground/PlaygroundTypes';
import { PlaybackStatus, RecordingStatus } from '../../features/sourceRecorder/SourceRecorderTypes';
import { WORKSPACE_BASE_PATHS } from '../../pages/fileSystem/createInBrowserFileSystem';
import { Assessment } from '../assessment/AssessmentTypes';
import { FileSystemState } from '../fileSystem/FileSystemTypes';
import Constants from '../utils/Constants';
import { createContext } from '../utils/JsSlangHelper';
import {
  DebuggerContext,
  WorkspaceLocation,
  WorkspaceManagerState,
  WorkspaceState
} from '../workspace/WorkspaceTypes';
import { ExternalLibraryName } from './types/ExternalTypes';
import { SessionState } from './types/SessionTypes';

export type OverallState = {
  readonly academy: AcademyState;
  readonly achievement: AchievementState;
  readonly application: ApplicationState;
  readonly playground: PlaygroundState;
  readonly session: SessionState;
  readonly workspaces: WorkspaceManagerState;
  readonly dashboard: DashboardState;
  readonly fileSystem: FileSystemState;
};

export type ApplicationState = {
  readonly environment: ApplicationEnvironment;
};

export type Story = {
  story: string;
  playStory: boolean;
};

export type GameState = {
  collectibles: { [id: string]: string };
  completed_quests: string[];
};

/**
 * An output while the program is still being run in the interpreter. As a
 * result, there are no return values or SourceErrors yet. However, there could
 * have been calls to display (console.log) that need to be printed out.
 */
export type RunningOutput = {
  type: 'running';
  consoleLogs: string[];
};

/**
 * An output which reflects the program which the user had entered. Not a true
 * Output from the interpreter, but simply there to let he user know what had
 * been entered.
 */
export type CodeOutput = {
  type: 'code';
  value: string;
};

/**
 * An output which represents a program being run successfully, i.e. with a
 * return value at the end. A program can have either a return value, or errors,
 * but not both.
 */
export type ResultOutput = {
  type: 'result';
  value: any;
  consoleLogs: string[];
  runtime?: number;
  isProgram?: boolean;
};

/**
 * An output which represents a program being run unsuccessfully, i.e. with
 * errors at the end. A program can have either a return value, or errors, but
 * not both.
 */
export type ErrorOutput = {
  type: 'errors';
  errors: SourceError[];
  consoleLogs: string[];
};

export type InterpreterOutput = RunningOutput | CodeOutput | ResultOutput | ErrorOutput;

export enum ApplicationEnvironment {
  Development = 'development',
  Production = 'production',
  Test = 'test'
}

export enum Role {
  Student = 'student',
  Staff = 'staff',
  Admin = 'admin'
}

/**
 * Defines the languages available for use on Source Academy,
 * including Source sublanguages and other languages e.g. full JS.
 * For external libraries, see ExternalLibrariesTypes.ts
 */
export interface SALanguage extends Language {
  displayName: string;
}

const variantDisplay: Map<Variant, string> = new Map([
  [Variant.TYPED, 'Typed'],
  [Variant.WASM, 'WebAssembly'],
  [Variant.NON_DET, 'Non-Det'],
  [Variant.CONCURRENT, 'Concurrent'],
  [Variant.LAZY, 'Lazy'],
  [Variant.GPU, 'GPU'],
  [Variant.NATIVE, 'Native'],
  [Variant.EXPLICIT_CONTROL, 'Explicit-Control']
]);

export const fullJSLanguage: SALanguage = {
  chapter: Chapter.FULL_JS,
  variant: Variant.DEFAULT,
  displayName: 'full JavaScript'
};

export const fullTSLanguage: SALanguage = {
  chapter: Chapter.FULL_TS,
  variant: Variant.DEFAULT,
  displayName: 'full TypeScript'
};

export const htmlLanguage: SALanguage = {
  chapter: Chapter.HTML,
  variant: Variant.DEFAULT,
  displayName: 'HTML'
};

export const styliseSublanguage = (chapter: Chapter, variant: Variant = Variant.DEFAULT) => {
  switch (chapter) {
    case Chapter.FULL_JS:
      return fullJSLanguage.displayName;
    case Chapter.FULL_TS:
      return fullTSLanguage.displayName;
    case Chapter.HTML:
      return htmlLanguage.displayName;
    default:
      return `Source \xa7${chapter}${
        variantDisplay.has(variant) ? ` ${variantDisplay.get(variant)}` : ''
      }`;
  }
};

export const sublanguages: Language[] = [
  { chapter: Chapter.SOURCE_1, variant: Variant.DEFAULT },
  { chapter: Chapter.SOURCE_1, variant: Variant.TYPED },
  { chapter: Chapter.SOURCE_1, variant: Variant.WASM },
  { chapter: Chapter.SOURCE_1, variant: Variant.LAZY },
  { chapter: Chapter.SOURCE_1, variant: Variant.NATIVE },
  { chapter: Chapter.SOURCE_2, variant: Variant.DEFAULT },
  { chapter: Chapter.SOURCE_2, variant: Variant.TYPED },
  { chapter: Chapter.SOURCE_2, variant: Variant.LAZY },
  { chapter: Chapter.SOURCE_2, variant: Variant.NATIVE },
  { chapter: Chapter.SOURCE_3, variant: Variant.DEFAULT },
  { chapter: Chapter.SOURCE_3, variant: Variant.TYPED },
  { chapter: Chapter.SOURCE_3, variant: Variant.CONCURRENT },
  { chapter: Chapter.SOURCE_3, variant: Variant.NON_DET },
  { chapter: Chapter.SOURCE_3, variant: Variant.NATIVE },
  { chapter: Chapter.SOURCE_4, variant: Variant.DEFAULT },
  { chapter: Chapter.SOURCE_4, variant: Variant.TYPED },
  { chapter: Chapter.SOURCE_4, variant: Variant.GPU },
  { chapter: Chapter.SOURCE_4, variant: Variant.NATIVE },
  { chapter: Chapter.SOURCE_4, variant: Variant.EXPLICIT_CONTROL }
];

export const sourceLanguages: SALanguage[] = sublanguages.map(sublang => {
  return {
    ...sublang,
    displayName: styliseSublanguage(sublang.chapter, sublang.variant)
  };
});

export const defaultLanguages = sourceLanguages.filter(
  sublang => sublang.variant === Variant.DEFAULT
);

export const variantLanguages = sourceLanguages.filter(
  sublang => sublang.variant !== Variant.DEFAULT
);

export const isSourceLanguage = (chapter: Chapter) =>
  [Chapter.SOURCE_1, Chapter.SOURCE_2, Chapter.SOURCE_3, Chapter.SOURCE_4].includes(chapter);

const currentEnvironment = (): ApplicationEnvironment => {
  switch (process.env.NODE_ENV) {
    case 'development':
      return ApplicationEnvironment.Development;
    case 'production':
      return ApplicationEnvironment.Production;
    default:
      return ApplicationEnvironment.Test;
  }
};

export const defaultAcademy: AcademyState = {
  gameCanvas: undefined
};

export const defaultApplication: ApplicationState = {
  environment: currentEnvironment()
};

export const defaultDashboard: DashboardState = {
  gradingSummary: {
    cols: [],
    rows: []
  }
};

export const defaultAchievement: AchievementState = {
  achievements: [],
  goals: [],
  users: [],
  assessmentOverviews: []
};

export const defaultPlayground: PlaygroundState = {
  githubSaveInfo: { repoName: '', filePath: '' }
};

export const defaultEditorValue = '// Type your program in here!';

/**
 * Create a default IWorkspaceState for 'resetting' a workspace.
 * Takes in parameters to set the js-slang library and chapter.
 *
 * @param workspaceLocation the location of the workspace, used for context
 */
export const createDefaultWorkspace = (workspaceLocation: WorkspaceLocation): WorkspaceState => ({
  autogradingResults: [],
  context: createContext<WorkspaceLocation>(
    Constants.defaultSourceChapter,
    [],
    workspaceLocation,
    Constants.defaultSourceVariant
  ),
  isFolderModeEnabled: false,
  activeEditorTabIndex: 0,
  editorTabs: [
    {
      filePath: ['playground', 'sicp'].includes(workspaceLocation)
        ? getDefaultFilePath(workspaceLocation)
        : undefined,
      value: ['playground', 'sourcecast', 'githubAssessments'].includes(workspaceLocation)
        ? defaultEditorValue
        : '',
      highlightedLines: [],
      breakpoints: []
    }
  ],
  programPrependValue: '',
  programPostpendValue: '',
  editorSessionId: '',
  isEditorReadonly: false,
  editorTestcases: [],
  externalLibrary: ExternalLibraryName.NONE,
  execTime: 1000,
  output: [],
  replHistory: {
    browseIndex: null,
    records: [],
    originalValue: ''
  },
  replValue: '',
  sharedbConnected: false,
  stepLimit: 1000,
  globals: [],
  isEditorAutorun: false,
  isRunning: false,
  isDebugging: false,
  enableDebugging: true,
  debuggerContext: {} as DebuggerContext
});

const defaultFileName = 'program.js';
export const getDefaultFilePath = (workspaceLocation: WorkspaceLocation) =>
  `${WORKSPACE_BASE_PATHS[workspaceLocation]}/${defaultFileName}`;

export const defaultWorkspaceManager: WorkspaceManagerState = {
  assessment: {
    ...createDefaultWorkspace('assessment'),
    currentAssessment: undefined,
    currentQuestion: undefined,
    hasUnsavedChanges: false
  },
  grading: {
    ...createDefaultWorkspace('grading'),
    currentSubmission: undefined,
    currentQuestion: undefined,
    hasUnsavedChanges: false
  },
  playground: {
    ...createDefaultWorkspace('playground'),
    usingSubst: false,
    activeEditorTabIndex: 0,
    editorTabs: [
      {
        filePath: getDefaultFilePath('playground'),
        value: defaultEditorValue,
        highlightedLines: [],
        breakpoints: []
      }
    ]
  },
  sourcecast: {
    ...createDefaultWorkspace('sourcecast'),
    audioUrl: '',
    codeDeltasToApply: null,
    currentPlayerTime: 0,
    description: null,
    inputToApply: null,
    playbackData: {
      init: {
        editorValue: '',
        chapter: Chapter.SOURCE_1,
        externalLibrary: ExternalLibraryName.NONE
      },
      inputs: []
    },
    playbackDuration: 0,
    playbackStatus: PlaybackStatus.paused,
    sourcecastIndex: null,
    title: null,
    uid: null
  },
  sourcereel: {
    ...createDefaultWorkspace('sourcereel'),
    playbackData: {
      init: {
        editorValue: '',
        chapter: Chapter.SOURCE_1,
        externalLibrary: ExternalLibraryName.NONE
      },
      inputs: []
    },
    recordingStatus: RecordingStatus.notStarted,
    timeElapsedBeforePause: 0,
    timeResumed: 0
  },
  sicp: {
    ...createDefaultWorkspace('sicp'),
    usingSubst: false,
    activeEditorTabIndex: 0,
    editorTabs: [
      {
        filePath: getDefaultFilePath('sicp'),
        value: defaultEditorValue,
        highlightedLines: [],
        breakpoints: []
      }
    ]
  },
  githubAssessment: {
    ...createDefaultWorkspace('githubAssessment'),
    hasUnsavedChanges: false
  }
};

export const defaultSession: SessionState = {
  courses: [],
  group: null,
  gameState: {
    completed_quests: [],
    collectibles: {}
  },
  xp: 0,
  allUserXp: undefined,
  story: {
    story: '',
    playStory: false
  },
  assessments: new Map<number, Assessment>(),
  assessmentOverviews: undefined,
  agreedToResearch: undefined,
  sessionId: Date.now(),
  githubOctokitObject: { octokit: undefined },
  gradingOverviews: undefined,
  gradings: new Map<number, Grading>(),
  notifications: []
};

export const defaultFileSystem: FileSystemState = {
  inBrowserFileSystem: null
};

export const defaultState: OverallState = {
  academy: defaultAcademy,
  achievement: defaultAchievement,
  application: defaultApplication,
  dashboard: defaultDashboard,
  playground: defaultPlayground,
  session: defaultSession,
  workspaces: defaultWorkspaceManager,
  fileSystem: defaultFileSystem
};
