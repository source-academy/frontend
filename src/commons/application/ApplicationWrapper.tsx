import { useMemo } from 'react';
import { useDispatch } from 'react-redux';
import { RouterProvider } from 'react-router';
import { createBrowserRouter } from 'react-router-dom';

import { getFullAcademyRouterConfig, playgroundOnlyRouterConfig } from '../../routes/routerConfig';
import Constants from '../utils/Constants';
import { useSession } from '../utils/Hooks';
import { updateReactRouter } from './actions/CommonsActions';

/**
 * Application wrapper component which figures out which deployment and set of routes to render.
 *
 * There are 3 main types of deployments as follows:
 * 1. Playground-only (stripped-down backendless version of SA - e.g. https://sourceacademy.org)
 * 2. Full Academy (full SA to be deployed and configured with the backend - e.g. https://sourceacademy.nus.edu.sg)
 * 3. Disabled (disabled SA which only allows `staff` and `admin` accounts to log in - e.g. during examinations)
 */
const ApplicationWrapper: React.FC = () => {
  const dispatch = useDispatch();
  const { isLoggedIn, role, name, courseId } = useSession();

  const router = useMemo(() => {
    const routerConfig = Constants.playgroundOnly
      ? playgroundOnlyRouterConfig
      : getFullAcademyRouterConfig({
          name,
          role,
          isLoggedIn,
          courseId
        });

    const r = createBrowserRouter(routerConfig);
    dispatch(updateReactRouter(r));

    return r;
  }, [isLoggedIn, role, name, courseId, dispatch]);

  return <RouterProvider router={router} />;
};

export default ApplicationWrapper;
