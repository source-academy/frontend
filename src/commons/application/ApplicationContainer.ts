import { connect, MapDispatchToProps, MapStateToProps } from 'react-redux';
import { withRouter } from 'react-router';
import { bindActionCreators, Dispatch } from 'redux';
import { createCourse } from 'src/features/academy/AcademyActions';

import { logOut } from './actions/CommonsActions';
import {
  fetchUserAndCourse,
  loginGitHub,
  logoutGitHub,
  updateCourseResearchAgreement,
  updateLatestViewedCourse
} from './actions/SessionActions';
import Application, { DispatchProps, StateProps } from './Application';
import { OverallState } from './ApplicationTypes';

/**
 * Provides the title of the application for display.
 * An object with the relevant properties must be
 * returned instead of an object of type @type {IApplicationProps},
 * as the routing properties of @type {RouteComponentProps} are
 * provided using the withRouter() method below.
 */
const mapStateToProps: MapStateToProps<StateProps, {}, OverallState> = state => ({
  role: state.session.role,
  name: state.session.name,
  courses: state.session.courses,
  courseId: state.session.courseId,
  courseShortName: state.session.courseShortName,
  enableAchievements: state.session.enableAchievements,
  enableSourcecast: state.session.enableSourcecast,
  assessmentTypes: state.session.assessmentConfigurations?.map(e => e.type),
  agreedToResearch: state.session.agreedToResearch
});

const mapDispatchToProps: MapDispatchToProps<DispatchProps, {}> = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      handleLogOut: logOut,
      handleGitHubLogIn: loginGitHub,
      handleGitHubLogOut: logoutGitHub,
      fetchUserAndCourse: fetchUserAndCourse,
      updateLatestViewedCourse: updateLatestViewedCourse,
      handleCreateCourse: createCourse,
      updateCourseResearchAgreement: updateCourseResearchAgreement
    },
    dispatch
  );

const ApplicationContainer = withRouter(connect(mapStateToProps, mapDispatchToProps)(Application));

export default ApplicationContainer;
