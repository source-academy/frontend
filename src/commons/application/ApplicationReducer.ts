import { Action, Reducer } from 'redux';

import { ApplicationState, defaultApplication } from './ApplicationTypes';

export const ApplicationReducer: Reducer<ApplicationState> = (
  state = defaultApplication,
  action: Action
) => {
  return state;
};
