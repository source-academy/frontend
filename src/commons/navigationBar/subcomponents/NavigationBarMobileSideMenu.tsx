import { Classes, Drawer, Icon } from '@blueprintjs/core';
import { IconNames } from '@blueprintjs/icons';
import classNames from 'classnames';
import * as React from 'react';
import { NavLink } from 'react-router-dom';
import { Role } from 'src/commons/application/ApplicationTypes';
import Constants from 'src/commons/utils/Constants';
import { assessmentTypeLink } from 'src/commons/utils/ParamParseHelper';

import { AssessmentType } from '../../assessment/AssessmentTypes';
import NotificationBadgeContainer from '../../notificationBadge/NotificationBadgeContainer';
import { filterNotificationsByType } from '../../notificationBadge/NotificationBadgeHelper';
import { icons } from './AcademyNavigationBar';

type NavigationBarMobileSideMenuProps = DrawerProps & OwnProps;

type DrawerProps = {
  name?: string;
  role?: Role;
  isOpen: boolean;
  onClose: () => void;
  handleGitHubLogIn: () => void;
  handleGitHubLogOut: () => void;
};

type OwnProps = {
  enableAchievements?: boolean;
  enableSourcecast?: boolean;
  assessmentTypes?: AssessmentType[];
  courseId?: number;
};

const NavigationBarMobileSideMenu: React.FC<NavigationBarMobileSideMenuProps> = props => (
  <Drawer
    isOpen={props.isOpen}
    position="left"
    onClose={props.onClose}
    title=""
    className={Classes.DARK}
  >
    {props.role ? (
      props.assessmentTypes?.map((assessmentType, idx) => (
        <NavLink
          to={`/courses/${props.courseId}/${assessmentTypeLink(assessmentType)}`}
          activeClassName={Classes.ACTIVE}
          className={classNames(
            'NavigationBar__link__mobile',
            Classes.BUTTON,
            Classes.MINIMAL,
            Classes.LARGE
          )}
          onClick={props.onClose}
          key={idx}
        >
          <Icon icon={icons[idx]} />
          <div>{assessmentType}</div>
          <NotificationBadgeContainer
            notificationFilter={filterNotificationsByType(assessmentType)}
            disableHover={true}
          />
        </NavLink>
      ))
    ) : (
      <></>
    )}

    {props.role && props.enableSourcecast && (
      <NavLink
        activeClassName={Classes.ACTIVE}
        className={classNames(
          'NavigationBar__link__mobile',
          Classes.BUTTON,
          Classes.MINIMAL,
          Classes.LARGE
        )}
        to={`/courses/${props.courseId}/sourcecast`}
        onClick={props.onClose}
      >
        <Icon icon={IconNames.MUSIC} />
        <div>Sourcecast</div>
      </NavLink>
    )}

    {props.role && (
      <NavLink
        activeClassName={Classes.ACTIVE}
        className={classNames(
          'NavigationBar__link__mobile',
          Classes.BUTTON,
          Classes.MINIMAL,
          Classes.LARGE
        )}
        to="/playground"
        onClick={props.onClose}
      >
        <Icon icon={IconNames.CODE} />
        <div>Playground</div>
      </NavLink>
    )}

    {Constants.enableGitHubAssessments && (
      <NavLink
        activeClassName={Classes.ACTIVE}
        className={classNames(
          'NavigationBar__link_mobile',
          Classes.BUTTON,
          Classes.MINIMAL,
          Classes.LARGE
        )}
        to="/githubassessments"
        onClick={props.onClose}
      >
        <Icon icon={IconNames.BRIEFCASE} />
        <div className="navbar-button-text">Classroom</div>
      </NavLink>
    )}
    {props.name && (
      <NavLink
        activeClassName={Classes.ACTIVE}
        className={classNames(
          'NavigationBar__link_mobile',
          Classes.BUTTON,
          Classes.MINIMAL,
          Classes.LARGE
        )}
        to={`/sicpjs/`}
        onClick={props.onClose}
      >
        <Icon icon={IconNames.BOOK} />
        <div className="navbar-button-text">SICP JS</div>
      </NavLink>
    )}
    {props.role && props.enableAchievements && (
      <NavLink
        activeClassName={Classes.ACTIVE}
        className={classNames(
          'NavigationBar__link__mobile',
          Classes.BUTTON,
          Classes.MINIMAL,
          Classes.LARGE
        )}
        to={`/courses/${props.courseId}/achievements`}
        onClick={props.onClose}
      >
        <Icon icon={IconNames.MOUNTAIN} />
        <div>Achievements</div>
      </NavLink>
    )}
  </Drawer>
);

export default NavigationBarMobileSideMenu;
