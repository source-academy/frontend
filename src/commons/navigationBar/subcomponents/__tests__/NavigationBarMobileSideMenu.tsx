import { shallow } from 'enzyme';

import { Role } from '../../../application/ApplicationTypes';
import NavigationBarMobileSideMenu from '../NavigationBarMobileSideMenu';

test('NavigationBarMobileSideMenu renders "Not logged in" correctly', () => {
  const props = {
    isOpen: true,
    onClose: () => {},
    handleGitHubLogIn: () => {},
    handleGitHubLogOut: () => {},
    assessmentTypes: [],
    courseId: 1
  };
  const tree = shallow(<NavigationBarMobileSideMenu {...props} />);
  expect(tree.debug()).toMatchSnapshot();
});

test('NavigationBarMobileSideMenu renders correctly when logged in (no course selected)', () => {
  const props = {
    isOpen: true,
    onClose: () => {},
    name: 'Avenger',
    handleGitHubLogIn: () => {},
    handleGitHubLogOut: () => {},
    assessmentTypes: [],
    courseId: 1
  };
  const tree = shallow(<NavigationBarMobileSideMenu {...props} />);
  expect(tree.debug()).toMatchSnapshot();
});

test('NavigationBarMobileSideMenu renders correctly when logged in (with course selected)', () => {
  const props = {
    isOpen: true,
    onClose: () => {},
    name: 'Avenger',
    role: Role.Student,
    handleGitHubLogIn: () => {},
    handleGitHubLogOut: () => {},
    assessmentTypes: ['Missions', 'Quests', 'Paths', 'Contests', 'Others'],
    courseId: 1
  };
  const tree = shallow(<NavigationBarMobileSideMenu {...props} />);
  expect(tree.debug()).toMatchSnapshot();
});
