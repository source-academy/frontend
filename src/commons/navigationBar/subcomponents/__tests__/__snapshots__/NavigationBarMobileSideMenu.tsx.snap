// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBarMobileSideMenu renders "Not logged in" correctly 1`] = `
"<Blueprint4.Drawer isOpen={true} position=\\"left\\" onClose={[Function: onClose]} title=\\"\\" className=\\"bp4-dark\\" canOutsideClickClose={true} style={{...}}>
  <NavLink activeClassName=\\"bp4-active\\" className=\\"NavigationBar__link_mobile bp4-button bp4-minimal bp4-large\\" to=\\"/githubassessments\\" onClick={[Function: onClose]}>
    <Blueprint4.Icon icon=\\"briefcase\\" />
    <div className=\\"navbar-button-text\\">
      Classroom
    </div>
  </NavLink>
</Blueprint4.Drawer>"
`;

exports[`NavigationBarMobileSideMenu renders correctly when logged in (no course selected) 1`] = `
"<Blueprint4.Drawer isOpen={true} position=\\"left\\" onClose={[Function: onClose]} title=\\"\\" className=\\"bp4-dark\\" canOutsideClickClose={true} style={{...}}>
  <NavLink activeClassName=\\"bp4-active\\" className=\\"NavigationBar__link_mobile bp4-button bp4-minimal bp4-large\\" to=\\"/githubassessments\\" onClick={[Function: onClose]}>
    <Blueprint4.Icon icon=\\"briefcase\\" />
    <div className=\\"navbar-button-text\\">
      Classroom
    </div>
  </NavLink>
  <NavLink activeClassName=\\"bp4-active\\" className=\\"NavigationBar__link_mobile bp4-button bp4-minimal bp4-large\\" to=\\"/sicpjs/\\" onClick={[Function: onClose]}>
    <Blueprint4.Icon icon=\\"book\\" />
    <div className=\\"navbar-button-text\\">
      SICP JS
    </div>
  </NavLink>
</Blueprint4.Drawer>"
`;

exports[`NavigationBarMobileSideMenu renders correctly when logged in (with course selected) 1`] = `
"<Blueprint4.Drawer isOpen={true} position=\\"left\\" onClose={[Function: onClose]} title=\\"\\" className=\\"bp4-dark\\" canOutsideClickClose={true} style={{...}}>
  <NavLink to=\\"/courses/1/missions\\" activeClassName=\\"bp4-active\\" className=\\"NavigationBar__link__mobile bp4-button bp4-minimal bp4-large\\" onClick={[Function: onClose]}>
    <Blueprint4.Icon icon=\\"flame\\" />
    <div>
      Missions
    </div>
    <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
  </NavLink>
  <NavLink to=\\"/courses/1/quests\\" activeClassName=\\"bp4-active\\" className=\\"NavigationBar__link__mobile bp4-button bp4-minimal bp4-large\\" onClick={[Function: onClose]}>
    <Blueprint4.Icon icon=\\"lightbulb\\" />
    <div>
      Quests
    </div>
    <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
  </NavLink>
  <NavLink to=\\"/courses/1/paths\\" activeClassName=\\"bp4-active\\" className=\\"NavigationBar__link__mobile bp4-button bp4-minimal bp4-large\\" onClick={[Function: onClose]}>
    <Blueprint4.Icon icon=\\"predictive-analysis\\" />
    <div>
      Paths
    </div>
    <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
  </NavLink>
  <NavLink to=\\"/courses/1/contests\\" activeClassName=\\"bp4-active\\" className=\\"NavigationBar__link__mobile bp4-button bp4-minimal bp4-large\\" onClick={[Function: onClose]}>
    <Blueprint4.Icon icon=\\"comparison\\" />
    <div>
      Contests
    </div>
    <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
  </NavLink>
  <NavLink to=\\"/courses/1/others\\" activeClassName=\\"bp4-active\\" className=\\"NavigationBar__link__mobile bp4-button bp4-minimal bp4-large\\" onClick={[Function: onClose]}>
    <Blueprint4.Icon icon=\\"manual\\" />
    <div>
      Others
    </div>
    <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
  </NavLink>
  <NavLink activeClassName=\\"bp4-active\\" className=\\"NavigationBar__link__mobile bp4-button bp4-minimal bp4-large\\" to=\\"/playground\\" onClick={[Function: onClose]}>
    <Blueprint4.Icon icon=\\"code\\" />
    <div>
      Playground
    </div>
  </NavLink>
  <NavLink activeClassName=\\"bp4-active\\" className=\\"NavigationBar__link_mobile bp4-button bp4-minimal bp4-large\\" to=\\"/githubassessments\\" onClick={[Function: onClose]}>
    <Blueprint4.Icon icon=\\"briefcase\\" />
    <div className=\\"navbar-button-text\\">
      Classroom
    </div>
  </NavLink>
  <NavLink activeClassName=\\"bp4-active\\" className=\\"NavigationBar__link_mobile bp4-button bp4-minimal bp4-large\\" to=\\"/sicpjs/\\" onClick={[Function: onClose]}>
    <Blueprint4.Icon icon=\\"book\\" />
    <div className=\\"navbar-button-text\\">
      SICP JS
    </div>
  </NavLink>
</Blueprint4.Drawer>"
`;
